sky/skypilot_config.py

```diff
@@ -93,8 +93,8 @@ def get_nested(keys: Sequence[str], default_value: Any) -> Any:
     return curr
 
 
-def pop_nested(keys: Sequence[str]) -> Dict[str, Any]:
-    """Returns a deep-copied config with the nested key popped.
+def set_nested(keys: Sequence[str], value: Any) -> Dict[str, Any]:
```

Comment:
Shall we write some unit tests?

I tried to test it out by having an config file
```
aws:
```
then
```
>>> from sky import skypilot_config
>>> skypilot_config.set_nested(('x',), None)
{'aws': None}
>>> skypilot_config.set_nested(('x','y'), None)
{'aws': None, 'x': {}}
```
which seems unexpected.